pkg/operator/ceph/object/spec.go

```diff
@@ -179,6 +180,28 @@ func (c *clusterConfig) makeRGWPodSpec(rgwConfig *rgwConfig) (v1.PodTemplateSpec
 			podSpec.Volumes = append(podSpec.Volumes, vaultFileVol, tmpvolume)
 			podSpec.InitContainers = append(podSpec.InitContainers,
 				c.vaultTokenInitContainer(rgwConfig))
+		} else if c.store.Spec.Security.KeyManagementService.IsK8sAuthEnabled() {
+			vaultFileVol, _ := kms.VaultVolumeAndMount(c.store.Spec.Security.KeyManagementService.ConnectionDetails,
+				"")
+			tmpvolume := v1.Volume{
+				Name: rgwVaultVolumeName,
+				VolumeSource: v1.VolumeSource{
+					EmptyDir: &v1.EmptyDirVolumeSource{},
+				},
+			}
+
+			podSpec.Volumes = append(podSpec.Volumes, vaultFileVol, tmpvolume)
+			vaultAgentVol := v1.Volume{Name: "vault-agent-config",
+				VolumeSource: v1.VolumeSource{
+					ConfigMap: &v1.ConfigMapVolumeSource{
+						LocalObjectReference: v1.LocalObjectReference{Name: "vault-agent-cm"},
+						Items:                []v1.KeyToPath{{Key: "vault-agent-config.hcl", Path: "vault-agent-config.hcl"}},
+					},
+				},
+			}
+			podSpec.Volumes = append(podSpec.Volumes, vaultAgentVol)
+			podSpec.Containers = append(podSpec.Containers,
+				c.vaultAgentContainer(rgwConfig))
```

Comment:
Can we do this as an init container instead of a sidecar to reduce resource usage?

Reply:
@BlaineEXE : Here RGW needs to send the request to vault-agent for fetching the key instead of vault-server so this need to up and running

Reply:
You are already setting resources for the sidecar here: https://github.com/rook/rook/pull/9872/files/5f71a8171c0048e777ebe60d24348cb7bbeb8d36#r827487726

But I am almost certain the vault agent doesn't need to have the same resource reservation as the RGW.

Reply:
Sure I will change that into minimal